.github/workflows/ci_tests_legacy.yaml

```diff
@@ -53,32 +52,19 @@ jobs:
           # fetch all history so that setuptools-scm works
           fetch-depth: 0
 
-      # Install Mambaforge with conda-forge dependencies
-      - name: Setup Mambaforge
-        uses: conda-incubator/setup-miniconda@v2.2.0
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@v15
         with:
-          activate-environment: pygmt
-          python-version: ${{ matrix.python-version }}
-          channels: conda-forge,nodefaults
-          channel-priority: strict
-          miniforge-version: latest
-          miniforge-variant: Mambaforge
-          mamba-version: "*"
-          run-post: false
-          use-mamba: true
-
-      # Install GMT and other required dependencies from conda-forge
-      - name: Install dependencies
-        run: |
-          mamba install gmt=${{ matrix.gmt_version }} numpy \
-                        pandas xarray netCDF4 packaging \
-                        contextily geopandas ipython rioxarray \
-                        build dvc make 'pytest>=6.0' \
-                        pytest-cov pytest-doctestplus pytest-mpl sphinx-gallery
+          environment-file: "ci/requirements/tests_legacy.yml"
```

Comment:
Could we reuse `ci/requirements/tests_with_optional_dependencies.yml` for the legacy tests and just set `gmt=6.3.0` in extra-specs? I think only the only difference is `pytest-cov`.

Reply:
Done in ece3db9e3b5e6da6efa17f7df79d6207a265d633.

Reply:
Also need to remove the `ci/requirements/tests_legacy.yml` file itself, and remove references from .github/ISSUE_TEMPLATE/bump_gmt_checklist.md

Reply:
> Also need to remove the `ci/requirements/tests_legacy.yml` file itself, 

It's already removed in commit ece3db9e3b5e6da6efa17f7df79d6207a265d633.

> and remove references from .github/ISSUE_TEMPLATE/bump_gmt_checklist.md

The file that is referenced in the "bump_gmt_checklist.md" is `.github/workflows/ci_tests_legacy.yaml`, not `ci/requirements/tests_legacy.yml`.

Reply:
Oh right, didn't look closely enough!